Automattic/jetpack
# jetpack: Fix random redirect module on author archives

#### Changes proposed in this Pull Request:

The "Set author name if we're on an author archive" bit doesn't seem
likely to have ever worked.

* It would have had a potential SQL injection, since it built an SQL
  fragment without any escaping.
* But it put the SQL fragment into the SQL as a string rather than as an
  SQL fragment, so instead it just didn't work.
  * Rather than being an SQL syntax error, it seems to have accidentally
	taken advantage of a weird MySQL feature where it will concatenate
	adjacent string literals.
* The field it tried to check against doesn't seem to actually be
  present in the tables in the query.

This fixes all of that. And rearranges the code to move some data
fetches after the checks for whether the module needs to run at all,
which might improve performance by a few nanoseconds.

#### Jetpack product discussion


p1649791995565429-slack-G78B6FYE7

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Try the feature, particularly by visiting an author page then appending `?random` to the URL. Make sure it returns a different random page each time, not the same one every time like it did before.
